src/validation.py

```diff
@@ -40,8 +40,19 @@ def validate_year(year):
         raise UserWarning(year_warning)
 
 
-def check_allocated_gf_matches_input_gf(pudl_out, gen_fuel_allocated):
-    """Checks that the allocated generation and fuel from EIA-923 matches the input totals."""
+def check_allocated_gf_matches_input_gf(
+    pudl_out, gen_fuel_allocated, threshold_percent=0.001
```

Comment:
If floating point errors are the concern here, the threshold should probably be smaller, though the exact size of expected error depends on the operations. python's isclose default relative error is 1e-9.